src/v/coproc/tests/coproc_test_fixture.cc

```diff
@@ -98,36 +78,64 @@ ss::future<coproc_test_fixture::opt_reader_data_t> coproc_test_fixture::drain(
     });
 }
 
+void interpret_batch(
```

Comment:
This is a little awkward.

It might be nicer as a `struct state {size_t;  model::offset; data_t;};`, and a `BinaryOperation` to `std::accumulate`

Reply:
I think I can reduce the amount of code by calling a utility method `sum_records` then this method just boils down to moving out the batches, so i can remove this method entirely